### PR TITLE
App layout design programmatically using Matlab

### LINKED ISSUE #6406: App layout design programmatically using Matlab
### Topic Suggestion
App layout design programmatically using Matlab

### Writing sample(s): 
https://www.section.io/engineering-education/image-processing-using-matlab/

#### Proposed article introduction
The app layout is the structure that interfaces the user with the app. The app consists of figures, these figures contain one or more UI(user interface). The UI components hold the interactive parts used while operating the app.

In Matlab, apps can be designed using GUI(graphical user interface) or programmatically using Matlab functions. This article will discuss designing app figure layout programmatically using Matlab functions. Through these functions, app UI components like buttons, dialog boxes, and containers can be made.

Apps are used in day-to-day activities in making life easier, they can be used in communication, data processing, and entertainment purposes.

#### Key takeaways 
- Creating app layout using matlab functions.
- Understanding the basic components of uifigure based apps.
- Understanding of uifigure function operations.

#### Article quality
The most obvious way of designing an app layout in matlab is by using the App designer or GUI(graphical user interface). This article discusses designing app layout through coding using matlab functions.

### PR DESCRIPTION
**Pull request by vitalisodhis**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6406 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/app-layout-programmatically-using-matlab/index.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
+### App layout design programmatically using Matlab
+
+### Introduction
+App layout consists of structures that contain the user interface(UI) elements. User interface elements fall into four main categories: Input control elements, informational elements, navigation components, and containers.
+
+Matlab provides an interactive environment for developing uifigure based apps layouts programmatically using Matlab functions.
```
Comment (jerimkaura) [inline content/articles/app-layout-programmatically-using-matlab/index.md:6]: Explain what UIfigure is to the reader.

File: content/articles/app-layout-programmatically-using-matlab/index.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+- Proper understanding of [MATLAB basics](/engineering-education/getting-started-with-Matlab/).
+
+### Objectives
+In the article, we discuss the design of the following layout structures of uifigure based apps :
+
+- Containers.
+- Components.
+- Dialog and notification boxes.
+
+### Containers
```
Comment (jerimkaura) [inline content/articles/app-layout-programmatically-using-matlab/index.md:22]: Use level 4 heading for sub headings.

File: content/articles/app-layout-programmatically-using-matlab/index.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+Containers are the structures that hold the app components together. For example, a default user interface container in Matlab is made using the function `uifigure`.
+
+Properties like the title of the container is added by using the sysntax `fig=uifigure('Name','figure title');` while the position is added using the function `fig.position=[];`.
+
+The below codes demonstrate creating of a default container. The title will be specified as 'Broadsheet 1'.
```
Comment (jerimkaura) [inline content/articles/app-layout-programmatically-using-matlab/index.md:27]: Use 'Code' or 'code snippet'. I don't think there's is 'codes'

File: content/articles/app-layout-programmatically-using-matlab/index.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+```
+![Container](/engineering-education/app-layout-design-prorammatically-using-matlab/applayout_a.jpg)
+
+### Types of UIfigure container
+Types of containers include:
+
+- Panel container.
+- Tab group container.
+
+### Panel container
```
Comment (jerimkaura) [inline content/articles/app-layout-programmatically-using-matlab/index.md:40]: Use level 4 heading to sub-headings for instance `####`

File: content/articles/app-layout-programmatically-using-matlab/index.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+Panel containers are used for grouping together the user interface components. Panel containers are made using function `uipanel();`.
+
+Panel container properties can be specified using the function `uipanel(Name, value);` the value is defined using one or more arguments. The function `parent` creates a panel in a specified parent container. The syntax for parent container properties is `panel=uipanel(parent,Name,value);`.
+
+The below codes shows the creation of a figure containing two panels and a button. The main panel is labelled 'panel 1', the sub-panel is labelled 'panel 2', and the button is labelled 'Start'.
```
Comment (jerimkaura) [inline content/articles/app-layout-programmatically-using-matlab/index.md:45]: Code.

File: content/articles/app-layout-programmatically-using-matlab/index.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+msg = 'Do you want to save this file.'; % confirmation dialog box massage 
+title ='Quit'; % box title
+selections= uiconfirm(fig,msg,title, ...
+           'Options',{'save','do not save','Cancel'}, ...
+           'DefaultOption',2,'CancelOption',3); % options available for selection
+```
+![Confirmation dialog box](/engineering-education/app-layout-design-prorammatically-using-matlab/applayout_l.jpg)
+
+### Conclusion
+App layout design is used in making apps or designing websites. In addition, apps are useful in communications, data storage, or entertainment. As demonstrated in the article, Matlab provides an interactive environment for designing app layout via coding using Matlab functions.
```
Comment (jerimkaura) [inline content/articles/app-layout-programmatically-using-matlab/index.md:195]: Make this statement  clear.

**Review by jerimkaura — CHANGES_REQUESTED** [review]
Great article @vitalisodhis 

However, consider working on the changes mentioned.

**Comment by vitalisodhis** [comment]
@jerimkaura  Changes made.

**Review by jerimkaura — COMMENTED** [review]
Hey @vitalisodhis 

Make sure to be going through your article before submission. I had to make major changes over:

- Punctuation
- Grammar mistakes
- Spelling
- Sentence structure

Check on the changes I made.

**Review by jerimkaura — APPROVED** [review]
Moving to the final review.

**Pull request merged**